### DISCUSSION TITLE
Can GitHub Codespaces be used as a development environment for Retina?

### DISCUSSION THREAD

**Discussion by roopeshsn** [microsoft/retina#348] [Q&A] [answered]
I would like to get community views on using codespaces as a dev env for Retina.

**Comment by SRodi** [comment]
@xiaozhiche320

**Comment by xiaozhiche320** [comment, marked as answer]
Hi Roopeshsn, I’m happy to answer your question.

Yes, you can definitely use Codespace, and we've already set up the dev configuration file in the .devcontainer folder. It’s straightforward to use Codespace as a development environment, and Kubectl is also done there. You just need to enable GPG signing in your Codespace config, create a fork from the main repository, and then use your fork to post any PRs.

FYI, Azure CLI, Prometheus, Grafana, and Hubble CLI might be needed depending on your use case, but they’re also easy to install.

Hope you enjoy the experience!

**Reply by SRodi** [reply]
Our doc also explains how to set up GPG signing for Codespaces users https://retina.sh/docs/Contributing/developing#cryptographic-signing-of-commits
